src/coordsim/trace_processor/trace_processor.py

```diff
@@ -9,25 +10,38 @@ class TraceProcessor():
     Trace processor class
     """
 
-    def __init__(self, params: SimulatorParams, env: Environment, trace: list,):
+    def __init__(self, params: SimulatorParams, env: Environment, trace: list, simulator: FlowSimulator):
         self.params = params
         self.env = env
         self.trace_index = 0
         self.trace = trace
-        self.env.process(self.change_inter_arrival_time())
+        self.simulator = simulator
+        self.env.process(self.process_trace())
 
-    def change_inter_arrival_time(self):
+    def process_trace(self):
         """
         Changes the inter arrival mean during simulation
         The initial time is read from the the config file, so if the inter_arrival_time set in the trace CSV
         file does not start from 0, then the simulator will use the value set in sim_config
 
         """
         self.timeout = float(self.trace[self.trace_index]['time']) - self.env.now
-        inter_arrival_mean = float(self.trace[self.trace_index]['inter_arrival_mean'])
+        inter_arrival_mean = self.trace[self.trace_index]['inter_arrival_mean']
         yield self.env.timeout(self.timeout)
         log.debug(f"Inter arrival mean changed to {inter_arrival_mean} at {self.env.now}")
-        self.params.inter_arr_mean = inter_arrival_mean
+        if 'node' in self.trace[self.trace_index]:
+            node_id = self.trace[self.trace_index]['node']
+            if inter_arrival_mean == 'None':
+                self.params.inter_arr_mean[node_id] = None
+            else:
+                inter_arrival_mean = float(inter_arrival_mean)
+                old_mean = self.params.inter_arr_mean[node_id]
+                self.params.inter_arr_mean[node_id] = inter_arrival_mean
+                if old_mean is None:
+                    self.simulator.generate_flow(node_id)
```

Comment:
Did you test it and it worked like this (eg, set ingress to None and then to another value)? I thought, you'd need to use `env.process()` here to not get stuck at this point, since the `generate_flow()` never returns.

If it works, feel free to merge. I don't have any more complaints! :)

Reply:
I totally forgot about this, good catch.